graphql-code-generator/src/main/java/io/ballerina/graphql/generator/service/combiner/ServiceFileCombiner.java

```diff
@@ -0,0 +1,124 @@
+
+        NodeList<ModuleMemberDeclarationNode> allMembers = createNodeList(moduleMembers);
+        ModulePartNode contentNode =
+                createModulePartNode(prevContentNode.imports(), allMembers, createToken(SyntaxKind.EOF_TOKEN));
+
+        TextDocument textDocument = TextDocuments.from(CodeGeneratorConstants.EMPTY_STRING);
+        SyntaxTree syntaxTree = SyntaxTree.from(textDocument);
+        return syntaxTree.modifyWith(contentNode);
+    }
+
+    private List<ModuleMemberDeclarationNode> generateNewMembers(NodeList<ModuleMemberDeclarationNode> prevMembers,
+                                                                 NodeList<ModuleMemberDeclarationNode> nextMembers) {
+        List<ModuleMemberDeclarationNode> newMembers = new ArrayList<>();
+        HashMap<ModuleMemberDeclarationNode, Boolean> nextMemberAvailable = new HashMap<>();
+        for (ModuleMemberDeclarationNode nextMember : nextMembers) {
+            nextMemberAvailable.put(nextMember, false);
+        }
+        for (ModuleMemberDeclarationNode prevMember : prevMembers) {
+            boolean foundMatch = false;
+            for (ModuleMemberDeclarationNode nextMember : nextMembers) {
+                if (isMemberEquals(prevMember, nextMember)) {
+                    foundMatch = true;
+                    nextMemberAvailable.put(nextMember, true);
+                    break;
+                }
+            }
+            if (!foundMatch) {
+                addMemberToRespectiveMemberList(prevMember);
+            }
+        }
+        for (Map.Entry<ModuleMemberDeclarationNode, Boolean> availableEntry : nextMemberAvailable.entrySet()) {
+            Boolean available = availableEntry.getValue();
+            if (!available) {
+                ModuleMemberDeclarationNode notAvailableMember = availableEntry.getKey();
+                if (notAvailableMember instanceof ServiceDeclarationNode) {
+                    moduleServiceDeclarations.add(notAvailableMember);
+                }
+            }
+        }
+        return newMembers;
+    }
+
+    private void addMemberToRespectiveMemberList(ModuleMemberDeclarationNode member) {
+        if (member instanceof ModuleVariableDeclarationNode) {
+            moduleVariables.add(member);
+        } else if (member instanceof ServiceDeclarationNode) {
+            moduleServiceDeclarations.add(member);
+        }
+    }
+
+    private boolean isMemberEquals(ModuleMemberDeclarationNode prevMember, ModuleMemberDeclarationNode nextMember) {
+        if (prevMember instanceof ServiceDeclarationNode && nextMember instanceof ServiceDeclarationNode) {
+            ServiceDeclarationNode prevServiceDeclaration = (ServiceDeclarationNode) prevMember;
+            ServiceDeclarationNode nextServiceDeclaration = (ServiceDeclarationNode) nextMember;
+            if (isServiceDeclarationMatch(prevServiceDeclaration, nextServiceDeclaration)) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
We can simplify this:

```suggestion
        if (prevMember.kind() != nextMember.kind() || prevMember.kind() != SyntaxKind.SERVICE_DECLARATION) {
            return false;
        }
        return isServiceDeclarationMatch((ServiceDeclarationNode) prevMember, (ServiceDeclarationNode) nextMember)
```